docs/source/reference/router/configuration.mdx

```diff
@@ -1145,6 +1147,17 @@ Before increasing this limit significantly consider testing performance
 in an environment similar to your production, especially if some clients are untrusted.
 Many concurrent large requests could cause the router to run out of memory.
 
+##### `experimental_http1_max_request_headers`
+
+Limit the maximum number of headers of incoming HTTP1 requests.
+The default value is 100 headers.
+
+If router receives more headers than the buffer size, it responds to the client with `431 Request Header Fields Too Large`.
+
+##### `experimental_http1_max_request_buf_size`
+
+Limit the maximum buffer size for the HTTP1 connection. Default is ~400kb.
+
```

Comment:
If we document this feature at all, I think we should also document that custom binary users need to copy the `[patch.crates-io]` section in their own `Cargo.toml`

Reply:
Added note to users of crate